portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -85,7 +48,7 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
         Random rand = new Random();
 
         response.setContentType("text/html;");
-        response.getWriter().println(countries.get(rand.nextInt(20)));
+        response.getWriter().println(countries.get(rand.nextInt(countries.size())));
```

Comment:
When `println(..)` is receiving a `Country`, there will be a conversion of this `Object` to a `String`. By default (`Object#toString()`) this prints the class name and the memory location of the object.

You should override the `toString()` method in your `Country` class if you're "printing" it in the response like this -- or format it in this method.

Reply:
To get both the country name and the capital, do I have to use two promises?